tests/test_network.py

```diff
@@ -0,0 +1,152 @@
+        ("POST", URL(f"{SUPERVISOR_URL}/network/reload"))
+    }
+
+
+async def test_network_interface_info(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test network interface info API."""
+    responses.get(
+        f"{SUPERVISOR_URL}/network/interface/end0/info",
+        status=200,
+        body=load_fixture("network_interface_info.json"),
+    )
+    result = await supervisor_client.network.interface_info("end0")
+    assert result.interface == "end0"
+    assert result.type == "ethernet"
+    assert result.enabled is True
+    assert result.mac == "00:11:22:33:44:55"
+    assert result.ipv4.method == "static"
+    assert result.ipv4.address[0].with_prefixlen == "192.168.1.2/24"
+    assert result.ipv4.nameservers[0].compressed == "192.168.1.1"
+    assert result.ipv4.gateway.compressed == "192.168.1.1"
+    assert result.ipv4.ready is True
+    assert result.ipv6.method == "disabled"
+    assert result.ipv6.address[0].with_prefixlen == "fe80::819d:c479:d712:7a77/64"
+    assert result.ipv6.gateway is None
+    assert result.wifi is None
+    assert result.vlan is None
+
+
+async def test_network_update_interface(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test network interface update API."""
+    responses.post(f"{SUPERVISOR_URL}/network/interface/end0/update", status=200)
+
+    # Must set something
+    with pytest.raises(TypeError):
+        config = NetworkInterfaceConfig()
+
+    config = NetworkInterfaceConfig(
+        ipv4=IPv4Config(
+            method=InterfaceMethod.STATIC,
+            address=[IPv4Interface("192.168.1.2/24")],
+            gateway=IPv4Address("192.168.1.1"),
+            nameservers=[IPv4Address("192.168.1.1")],
+        )
+    )
+    assert (
+        await supervisor_client.network.update_interface("end0", config=config) is None
+    )
+    assert responses.requests.keys() == {
+        ("POST", URL(f"{SUPERVISOR_URL}/network/interface/end0/update"))
+    }
+
+
+async def test_network_access_points(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test network access points API."""
+    responses.get(
+        f"{SUPERVISOR_URL}/network/interface/end0/accesspoints",
+        status=200,
+        body=load_fixture("network_access_points.json"),
+    )
+    result = await supervisor_client.network.access_points("end0")
+    assert result[0].mode == "infrastructure"
+    assert result[0].ssid == "UPC4814466"
+    assert result[0].frequency == 2462
+    assert result[0].signal == 47
+    assert result[0].mac == "AA:BB:CC:DD:EE:FF"
+    assert result[1].ssid == "VQ@35(55720"
+
+
+async def test_network_save_vlan(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test network save vlan API."""
+    responses.post(f"{SUPERVISOR_URL}/network/interface/end0/vlan/1", status=200)
+
+    # Must set something
+    with pytest.raises(TypeError):
```

Comment:
This should raise ValueError.

Reply:
Yea that's corrected after the other PR. Will adjust after rebase 👍